main/src/library/Files.flix

```diff
@@ -805,22 +702,17 @@ mod Files {
     ///
     pub def list(f: String): Result[IOError, List[String]] \ IO = region rc {
         try {
-            import java.io.File.listFiles(): Array[File, rc] \ IO;
-            import java.io.File.toPath(): Path \ IO;
-            import java.nio.file.Path.toString(): String \ IO as pathToString;
-
-            let javaList = new File(f) |> listFiles;
+            let javaList = unchecked_cast(new File(f).listFiles() as Array[File, rc]);
```

Comment:
This unchecked_cast was needed for some reason. Without, it should just have type `Array[File, IO]` which should work fine, but somehow not doing it messes up the type of `javaList` to have type `File` instead of `Array[File, ..]` after the typer (not during)